lib/TelemetrySender.js

```diff
@@ -1,34 +1,68 @@
 /* globals Services */
 
 const {Cu} = require("chrome");
+const simplePrefs = require("sdk/simple-prefs");
 Cu.import("resource://gre/modules/Services.jsm");
 Cu.importGlobalProperties(["fetch"]);
 
-function TelemetrySender(pingEndpoint) {
-  Services.obs.addObserver(this, "tab-session-complete");
-  this._pingEndpoint = pingEndpoint;
+const ENDPOINT_PREF = "telemetry.ping.endpoint";
+const TELEMETRY_PREF = "telemetry";
+
+function TelemetrySender() {
+  this.enabled = simplePrefs.prefs[TELEMETRY_PREF];
+  this._pingEndpoint = simplePrefs.prefs[ENDPOINT_PREF];
+  this._onPrefChange = this._onPrefChange.bind(this);
+  simplePrefs.on(ENDPOINT_PREF, this._onPrefChange);
+  simplePrefs.on(TELEMETRY_PREF, this._onPrefChange);
+  if (this.enabled) {
+    this.init();
+  }
 }
 
 TelemetrySender.prototype = {
-  observe: function(subject, topic, data) {
+  observe(subject, topic, data) {
     if (topic === "tab-session-complete") {
       this._sendPing(data);
     }
   },
 
-  _sendPing: function(data) {
-    fetch(this._pingEndpoint, {method: "POST", body: data}).then(function(response) {
+  _onPrefChange(subject, topic, data) {
```

Comment:
You are correct! I was expecting the observer interface. I guess I should write those tests.